File: Assets/scenes/warzonescene/script/MissileFlareBehaviour.js
#pragma strict

var sounds : AudioClip[];
var flightDuration : float;	//how long we fly for before explodering
var successful : boolean = false;

private var parts : ParticleSystem;
private var randomSound : int = 0;
private var creationTime : float;

private var dying : boolean = false;

private var randomRotation : Quaternion;

function Start () {
	parts = GetComponentInChildren.<ParticleSystem>();	
	randomSound = Random.Range(0,sounds.Length);
	creationTime = Time.fixedTime;
	randomRotation = Quaternion.Euler(Random.Range(-0.5,0.5), Random.Range(-0.5,0.5),0);
	
}

function Update () {
	if(creationTime + flightDuration < Time.fixedTime){
		if(dying == false){
			dying = true;
			explode();
		}
	} else {
		
		transform.rotation *= randomRotation;
		transform.Translate(0,0,5); 
		
	}
}

function explode(){
	if(parts == null){
		parts = GetComponentInChildren.<ParticleSystem>();
	}
	if(successful){
		parts.Play();
	//GetComponent.<TrailRenderer>().enabled =
		AudioSource.PlayClipAtPoint(sounds[randomSound], transform.position);
	}
	yield WaitForSeconds(6);
	Destroy(gameObject);
}

File: Assets/scenes/warzonescene/script/MissileFlareBehaviour.cs
using UnityEngine;
using System;
using System.Collections;


public class MissileFlareBehaviour:MonoBehaviour{
	
	public AudioClip[] sounds;
	public float flightDuration;	//how long we fly for before explodering
	public bool successful = false;
	
	ParticleSystem parts;
	int randomSound = 0;
	float creationTime;
	
	bool dying = false;
	
	Quaternion randomRotation;
	
	public void Start() {
		parts = GetComponentInChildren<ParticleSystem>();	
		randomSound = UnityEngine.Random.Range(0,sounds.Length);
		creationTime = Time.fixedTime;
		randomRotation = Quaternion.Euler(UnityEngine.Random.Range(-0.5f,0.5f), UnityEngine.Random.Range(-0.5f,0.5f),0.0f);
		
	}
	
	public void Update() {
		if(creationTime + flightDuration < Time.fixedTime){
			if(dying == false){
				dying = true;
				StartCoroutine(explode());
			}
		} else {
			
			transform.rotation *= randomRotation;
			transform.Translate(0.0f,0.0f,5.0f); 
			
		}
	}
	
	public IEnumerator explode(){
		if(parts == null){
			parts = GetComponentInChildren<ParticleSystem>();
		}
		if(successful){
			parts.Play();
		//GetComponent.<TrailRenderer>().enabled =
			UsefulShit.PlayClipAt(sounds[randomSound], transform.position);
		}
		yield return new WaitForSeconds(6.0f);
		Destroy(gameObject);
	}
}